jetbrains-core/src/software/aws/toolkits/jetbrains/core/AwsResourceCache.kt

```diff
@@ -6,64 +6,121 @@ package software.aws.toolkits.jetbrains.core
 import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.components.ServiceManager
 import com.intellij.openapi.project.Project
-import software.amazon.awssdk.services.lambda.LambdaClient
-import software.aws.toolkits.core.ToolkitClientManager
+import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.warn
 import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsManager
-import software.aws.toolkits.jetbrains.services.lambda.LambdaFunction
-import software.aws.toolkits.jetbrains.services.lambda.toDataClass
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
 import java.util.concurrent.ConcurrentHashMap
 
-// TODO to be replaced with an actual resource implementation
-
+/**
+ * Intended to prevent repeated unnecessary calls to AWS to understand resource state.
+ *
+ * Will cache responses from AWS by [AwsRegion]/[ToolkitCredentialsProvider] - generically applicable to any AWS call.
+ */
 interface AwsResourceCache {
-    fun lambdaFunctions(): CompletableFuture<List<LambdaFunction>>
+
+    /**
+     * Get a [resource] either by making a call or returning it from the cache if present and unexpired. Uses the currently [AwsRegion]
```

Comment:
drop `currently` before region

Reply:
Done

Reply:
I would put active before the [AwsRegion], current way reads very odd